ir/instr.cpp

```diff
@@ -277,18 +280,34 @@ StateValue BinOp::toSMT(State &s) const {
 
   case FAdd:
     val = a.fadd(b);
+    if (flags & NINF)
+      not_poison &= !(a.isInf()) && !(b.isInf()) && !(val.isInf());
+    if (flags & NNAN)
+      not_poison &= !(a.isNaN()) && !(b.isNaN()) && !(val.isNaN());
```

Comment:
also, this code probably wants to be moved to a function since there's a lot of duplication here and there are many more fast-math flags to be supported